HOWTO.adoc

```diff
@@ -0,0 +1,86 @@
+= How to Write Good Documentation
+
+This guide will attempt to help you avoid common documentation issues and help you get your point across as clearly as possible.
+
+== Writing Style
+
+=== Passive vs. Active Voice
+When describing a documentation item, use a passive voice.
+
+> `thing`: Does a lot of stuff.
+
+When writing a warning, note, or similar, use an active voice. Speak _to_ the reader.
+
+> Warning: Doing `this` will give you weird results. You should try to do `that` instead.
+
+=== Abbreviations
+When writing in _plain English_, **avoid abbreviations**. This also includes things like "&" instead of "and" and "@" instead of "at".
+
+When writing _code examples_, **label abbreviations sufficiently**, either by self-labeling (match the abbreviation to the value) or with a comment.
+
+Bad abbreviations:
+[source,lua]
+----
+local a = VoxelManip(p1, p2) -- BAD: "a" is not a good choice for this variable
+local emin, emax = a:read_from_map(p1, p2) -- BAD: What does the "e" mean?
+----
+
+Good abbreviations:
+[source,lua]
+----
+local vm = VoxelManip(p1, p2) -- GOOD: Self-labeled
+local emin, emax = vm:read_from_map(p1, p2) -- Emerged minimum and maximum positions (GOOD)
+----
+
+=== Grammar and Punctuation
+* Always capitalize proper names and first letters in plain text.
+    ** Capitalization in quotation marks is not required and usually discouraged. The goal of quotation in documentation is to mirror _exactly_.
+* End all plain text with a period. This does not apply to code examples.
+* Use the Oxford (or serial) comma. It helps reduce confusion.
+
+
+== Technical Detail
+This is software documentation, so technical explanations are expected. But it is important to know when and how much to explain technical things.
+
+=== Object Names
+When referring to an object, use it's full name, the accepted shorthand, or it's highest generic term.
+
+For example: An `ItemStack` should be referred to as "ItemStack", "stack", or "object", depending on the need. It should _not_ be referred to as "userdata" or "reference". Objects that have metatables should usually _not_ be referred to as "metatable".
```

Comment:
I don't like this mushy object definition that mixes tables with metatables and userdata. We should define our terminology elsewhere.

Reply:
We are trying to reduce excess detail, such as referring to something as a "metatable" which is awful and confusing to pretty much anyone. It's an object, and we dont really need to elaborate further for the purposes of this.

Reply:
The object term is way too generic and overloaded. I'd refer to objects as instances. The metatable is the "class".